test/powershell/Language/Scripting/NativeExecution/NativeCommandArguments.Tests.ps1

```diff
@@ -1,70 +1,84 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
-Describe "Native Command Arguments" -tags "CI" {
-    # When passing arguments to native commands, quoted segments that contain
-    # spaces need to be quoted with '"' characters when they are passed to the
-    # native command (or to bash or sh on Linux).
-    #
-    # This test checks that the proper quoting is occuring by passing arguments
-    # to the testexe native command and looking at how it got the arguments.
-    It "Should handle quoted spaces correctly" {
-        $a = 'a"b c"d'
-        $lines = testexe -echoargs $a 'a"b c"d' a"b c"d
-        $lines.Count | Should -Be 3
-        $lines[0] | Should -BeExactly 'Arg 0 is <ab cd>'
-        $lines[1] | Should -BeExactly 'Arg 1 is <ab cd>'
-        $lines[2] | Should -BeExactly 'Arg 2 is <ab cd>'
-    }
+foreach ( $argumentListValue in 0,1 ) {
+    $PSNativeApplicationUsesArgumentList = $argumentListValue
+    Describe "Native Command Arguments" -tags "CI" {
+        # When passing arguments to native commands, quoted segments that contain
+        # spaces need to be quoted with '"' characters when they are passed to the
+        # native command (or to bash or sh on Linux).
+        #
+        # This test checks that the proper quoting is occuring by passing arguments
+        # to the testexe native command and looking at how it got the arguments.
+        It "Should handle quoted spaces correctly (ArgumentList=${PSNativeApplicationUsesArgumentList})" {
+            $a = 'a"b c"d'
```

Comment:
which test case represents the MSI scenario on Windows and `git commit -m "This 'has' quotes"`

Reply:
i don't think that's currently broken in our current environment, but it's easy enough to add to this.